adapters/repos/db/migrator.go

```diff
@@ -690,3 +692,9 @@ func (m *Migrator) WaitForStartup(ctx context.Context) error {
 func (m *Migrator) Shutdown(ctx context.Context) error {
 	return m.db.Shutdown(ctx)
 }
+
+// UpdateIndex creates index if not exists and idempotent in nature.
+func (m *Migrator) UpdateIndex(ctx context.Context, class *models.Class, state *sharding.State) error {
+	// add class is idempotent
+	return m.AddClass(ctx, class, state)
```

Comment:
The function is replacing the current index with a new one.
However, it's not clear to me what should happen to the existing index, as it might be used elsewhere
Additionally, it's unclear what should happen if some properties, or shards are not present in the updates.
We might need assistance from DB team to explore the actual behaviour.